lib/network-layer/src/Cardano/Wallet/Network.hs

```diff
@@ -275,14 +276,20 @@ mapChainFollower fpoint12 fpoint21 ftip fblocks cf =
 -------------------------------------------------------------------------------}
 
 -- | Error while trying to send a transaction
-data ErrPostTx = ErrPostTxValidationError Text | ErrPostTxMempoolFull
+data ErrPostTx
+    = ErrPostTxValidationError Text
+    | ErrPostTxMempoolFull
+    | ErrPostTxEraUnsupported AnyCardanoEra
     deriving (Generic, Show, Eq)
 
 instance ToText ErrPostTx where
     toText = \case
         ErrPostTxValidationError msg -> msg
         ErrPostTxMempoolFull ->
             "mempool was full and refused posted transaction"
+        ErrPostTxEraUnsupported era ->
+            "submitted transaction is not compatible with the current era: "
+                <> T.pack (show era)
```

Comment:
Suggestion:

Instead of trying to do string interpolation here (which is awkward to test), I would propose to use a constant string that defers to a rich error object. (See comment above!)